src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,76 @@
+package ro.msg.learning.shop.controller;
+
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.exception.ProductException;
+import ro.msg.learning.shop.service.IProductService;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+@RestController
+public class ProductController {
+    @Autowired
+    private IProductService productService;
+
+    @GetMapping("/products")
+    @ResponseBody
+    public List<ProductDto> getAllProducts()
+    {
+        return productService.getAllProducts();
+    }
+
+    @GetMapping("/products/{id}")
+    public ResponseEntity<ProductDto> getProductById(@PathVariable Integer id)
+    {
+        try
+        {
+            ProductDto productDto = productService.getProductById(id);
+            return ResponseEntity.status(HttpStatus.OK).body(productDto);
+        }catch (ProductException e)

```

Comment:
A best practice to handle rest exceptions is to create an exception handler method in the controller or better, a **ControllerAdvice** for global error handling. By using a ControllerAdvice, you can use the exact same exception handling methods but apply them to the whole application as opposed to just one controller. 
Here is an useful documentation that might help you with the exception handling: https://www.baeldung.com/exception-handling-for-rest-with-spring